WeakAuras/Transmission.lua

```diff
@@ -730,454 +889,570 @@ local function scamCheck(codes, data)
-            else
-              tinsert(tooltip, {2, L["Trigger:"], (status_types[trigger.event] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
-            end
-            if(trigger.event == "Combat Log" and trigger.subeventPrefix and trigger.subeventSuffix) then
-              tinsert(tooltip, {2, L["Message type:"], (WeakAuras.subevent_prefix_types[trigger.subeventPrefix] or L["Undefined"]).." "..(WeakAuras.subevent_suffix_types[trigger.subeventSuffix] or L["Undefined"]), 1, 1, 1, 1, 1, 1});
-            end
-          else
-            tinsert(tooltip, {2, L["Trigger:"], L["Custom"], 1, 1, 1, 1, 1, 1});
-          end
+  -- match the parent/single aura (if no children)
+  local oldParent = WeakAuras.GetData(data.id)
+  if oldParent and not isParentMatch(oldParent, data) then oldParent = nil end
+  if not oldParent or oldParent.uid ~= data.uid then
+    if data.uid then
+      for id, existingData in pairs(WeakAurasSaved.displays) do
+        if isParentMatch(existingData, data) then
+          oldParent = existingData
+          break
         end
       end
+    else
+      return nil
     end
+  end
 
-    if (import and #tooltip > 30) then
-      -- Truncate the tooltip to ~25 auras if there are more than ~30
-      local size = #tooltip
-      tooltip[26] = {2, " ",  "[...]", 1, 1, 1, 1, 1, 1};
-      local nrOfChildren = children and #children or data.controlledChildren and #data.controlledChildren or 0
-      tooltip[27] = {1, string.format(L["%s total auras"], nrOfChildren), "", 1, 1, 1, 1};
-      for i = 28, size do
-        tooltip[i] = nil;
-      end
-    end
+  if not oldParent then return nil end
 
-    local hasDescription = data.desc and data.desc ~= "";
-    local hasUrl = data.url and data.url ~= "";
-    local hasVersion = data.version and data.version ~= "";
+  info.oldData[0] = oldParent
+  info.newData[0] = data
+  info.diffs[0] = WeakAuras.diff(oldParent, data)
+  if info.diffs[0] then
+    info.modified = info.modified + 1
+  end
 
-    if(hasDescription or hasUrl or hasVersion) then
-      tinsert(tooltip, {1, " "});
+  if children then
+    -- setup
+    info.deleted = #oldParent.controlledChildren
+    info.added = #children
+    local UIDtoID, IDtoData, IDtoIndex = {}, {}, {}
+    for index, oldChildID in ipairs(oldParent.controlledChildren) do
+      local oldChild = WeakAuras.GetData(oldChildID)
+      IDtoData[oldChildID] = oldChild
+      IDtoIndex[oldChildID] = index
+      if oldChild.uid and not UIDtoID[oldChild.uid] then
+        UIDtoID[oldChild.uid] = oldChildID
+      end
     end
 
-    if(hasDescription) then
-      tinsert(tooltip, {1, "\""..data.desc.."\"", 1, 0.82, 0, 1});
+    -- confirm uid matches, find tentative matches
+    local tentativeMatches, confirmedNewMatches, confirmedOldMatches = {}, {}, {}
+    for index, newChild in ipairs(children) do
+      if newChild.uid then
+        local matchedID = UIDtoID[newChild.uid]
+        if matchedID then
```

Comment:
Even if a new child has a uid, if there's no uid match, there should be a fallback to a id match.

Essentially the code of the else branch of the check 3 lines up must be done if there's no matchedId.